mapillary/models/geojson.py

```diff
@@ -0,0 +1,350 @@
+        """Initializing Geometry constructor"""
+
+        # Validate that the geojson passed is indeed a dictionary
+        if not isinstance(geometry, dict):
+
+            # Raise InvalidOptionError
+            InvalidOptionError(
+                # The parameter that caused the exception
+                param="Geometry.__init__.geometry",
+                # The invalid value passed
+                value=geometry,
+                # The keys that should be passed instead
+                options=["dict"],
+            )
+
+        # Setting the type of the selected geometry
+        self.type: str = geometry["type"]
+
+        # Setting the coordinates of the geometry
+        self.coordinates: list = geometry["coordinates"]
+
+    def to_dict(self):
+        """Return dictionary representation of the geometry"""
+
+        return {"type": self.type, "coordinates": self.coordinates}
+
+    def __str__(self):
+        """Return the informal string representation of the Geometry"""
+
+        return f"{{'type': {self.type}, 'coordinates': {self.coordinates}}}"
+
+    def __repr__(self):
+        """Return the formal string representation of the Geometry"""
+
+        return f"{{'type': {self.type}, 'coordinates': {self.coordinates}}}"
+
+
+class Feature:
+    """Representation for a feature in a feature list
+
+    :param geojson: The GeoJSON as the input
+    :type geojson: dict
+
+    '''
+    :raise InvalidOptionError: Raised when the geojson passed is the invalid type - not a dict
+    '''
+
+    :return: A class representation of the model
+    :rtype: <class 'mapillary.models.geojson.Feature'>
+    """
+
+    def __init__(self, feature: dict) -> None:
+        """Initializing Feature constructor"""
+
+        # Validate that the geojson passed is indeed a dictionary
+        if not isinstance(feature, dict):
+
+            # If not, raise `InvalidOptionError`
+            InvalidOptionError(
+                # The parameter that caused the exception
+                param="Feature.__init__.feature",
+                # The invalid value passed
+                value=feature,
+                # The type of value that should be passed instead
+                options=["dict"],
+            )
+
+        # Setting the type of the selected FeatureList
+        self.type = "Feature"
+
+        # Setting the `geometry` property
+        self.geometry = Geometry(feature["geometry"])
+
+        # Setting the `properties` property
+        self.properties = Properties(feature["properties"])
+
+    def to_dict(self) -> dict:
+        """Return the dictionary representation of the Feature"""
+
+        return {
+            "type": self.type,
+            "geometry": self.geometry.to_dict(),
+            "properties": self.properties.to_dict(),
+        }
+
+    def __str__(self) -> str:
+        """Return the informal string representation of the Feature"""
+
+        return (
+            f"{{"
+            f"'type': '{self.type}', "
```

Comment:
should this {self.type} be a string?

Reply:
It is assigned by default to `"Feature"` in line 177. Can this parameter have something else in instead?

Yes, it should be. I had to use use the single quotes here because it would like { 'type': Feature, 'geometry': ... } would be weird.